### PR TITLE
[CI] Test MathComp 2 in CI

### PR DESCRIPTION
**Pull request by proux01**
Keep a mathcomp_1 target for targets that are not (yet) ported to MC2: analysis, deriving, fcsl_pcm, mathcomp_word, jasmin and serapi_test

- [ ] ~Added / updated **test-suite**.~

- [ ] ~Added **changelog**.~
- [ ] ~Added / updated **documentation**.~
  
  - [ ] ~Documented any new / changed **user messages**.~
  - [ ] ~Updated **documented syntax** by running `make doc_gram_rsts`.~

- [ ] ~Opened **overlay** pull requests.~

<!-- Pointers to relevant developer documentation:

Contributing guide: https://github.com/coq/coq/blob/master/CONTRIBUTING.md

Test-suite: https://github.com/coq/coq/blob/master/test-suite/README.md

Changelog: https://github.com/coq/coq/blob/master/doc/changelog/README.md

Building the doc: https://github.com/coq/coq/blob/master/doc/README.md
Sphinx: https://github.com/coq/coq/blob/master/doc/sphinx/README.rst
doc_gram: https://github.com/coq/coq/blob/master/doc/tools/docgram/README.md

Overlays: https://github.com/coq/coq/blob/master/dev/ci/user-overlays/README.md

### CODE REVIEW

**Comment by proux01** [comment]
@coqbot run full ci

**Comment by proux01** [comment]
I don't understand the CI failure?

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -740,6 +740,11 @@ library:ci-math_classes:
 
 library:ci-mathcomp:
   extends: .ci-template-flambda
+  needs:
+  - plugin:ci-elpi_hb  # for Hierarchy Builder
```
Comment (SkySkimmer) [inline .gitlab-ci.yml:744]: https://github.com/coq/coq/blob/master/dev/ci/README-users.md#add-your-project-by-submitting-a-pull-request
>the needs list must include all transitive dependencies including build:base or build:edge+flambda depending on the switch you chose

**Comment by proux01** [comment]
@coqbot run full ci

**Comment by proux01** [comment]
Looks like we have an out of memory. I guess we'll have to fix that before having MathComp 2 in the CI here.

**Comment by proux01** [comment]
@coqbot run full ci

**Comment by proux01** [comment]
@coqbot run full ci

**Comment by proux01** [comment]
Thanks to the work on elpi, the performance of MathComp 2 is now reasonable enough that the CI is green (fiat-crypto and metacoq failure are unrelated). So I'm marking this as ready for review.

Note that Quickchick perfectly compiles with MathComp 2 but is currently kept with MathComp 1 because of the following painful conflict:
* ci-mathcomp (2) uses ci-elpi_hb which uses the stable version of menhir from the Docker image
* ci-http requires both ci-quickchick (hence ci-mathcomp) and ci-menhir which installs the dev version of coq-menhirlib, requiring the dev version of menhir

File: dev/ci/ci-basic-overlay.sh [hunk 2]
```diff
@@ -219,7 +213,7 @@ project equations "https://github.com/mattam82/Coq-Equations" "main"
 ########################################################################
 project elpi "https://github.com/LPCIC/coq-elpi" "coq-master"
 
-project hierarchy_builder "https://github.com/math-comp/hierarchy-builder" "coq-master"
+project hierarchy_builder "https://github.com/math-comp/hierarchy-builder" "master"
```
Comment (SkySkimmer) [inline dev/ci/ci-basic-overlay.sh:216]: Shouldn't hb coq-master be updated instead?

Comment (proux01) [reply dev/ci/ci-basic-overlay.sh:216]: The coq-master branch of HB is no longer useful, I take the opportunity of this PR to replace it (c.f. https://github.com/math-comp/hierarchy-builder/pull/381#issuecomment-1697253073 ), sorry I should have written it in the commit message.

Comment (SkySkimmer) [reply dev/ci/ci-basic-overlay.sh:216]: So HB master is tested with Coq master in its CI and will receive the overlays?

Comment (proux01) [reply dev/ci/ci-basic-overlay.sh:216]: HB master will receive the overlay, we will delete the coq-master branch and add Coq master to the CI of the master branch.

**Review by SkySkimmer — APPROVED** [review]
>add Coq master to the CI of the master branch

Please ensure this is done promptly :)

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Comment by proux01** [comment]
> > add Coq master to the CI of the master branch
> 
> Please ensure this is done promptly :)

I'm working on it, thanks!

**Pull request merged**